src/TLuaInterpreterMMCP.cpp

```diff
@@ -0,0 +1,409 @@
+
+    Host* pHost = &getHostFromLua(L);
+    if (!pHost->mmcpServer) {
+        pHost->initMMCPServer();
+    }
+
+    QPair<bool, QString> const result = pHost->mmcpServer->chatPrivate(target);
+
+    if (!result.first) {
+        return warnArgumentValue(L, __func__, result.second.toUtf8().constData());
+    }
+
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+int TLuaInterpreter::chatServe(lua_State* L)
+{
+    const QString target = getVerifiedString(L, __func__, 1, "target");
+
+    Host* pHost = &getHostFromLua(L);
+    if (!pHost->mmcpServer) {
+        pHost->initMMCPServer();
+    }
+
+    QPair<bool, QString> const result = pHost->mmcpServer->serve(target);
+
+    if (!result.first) {
+        return warnArgumentValue(L, __func__, result.second.toUtf8().constData());
+    }
+
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+int TLuaInterpreter::chatSetGroup(lua_State* L)
+{
+    const QString target = getVerifiedString(L, __func__, 1, "target");
+    const QString group = getVerifiedString(L, __func__, 2, "group");
+
+    Host* pHost = &getHostFromLua(L);
+    if (!pHost->mmcpServer) {
+        pHost->initMMCPServer();
+    }
+
+    QPair<bool, QString> const result = pHost->mmcpServer->chatSetGroup(target, group);
+
+    if (!result.first) {
+        return warnArgumentValue(L, __func__, result.second.toUtf8().constData());
+    }
+
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+int TLuaInterpreter::chatSideChannel(lua_State* L)
+{
+    const QString channel = getVerifiedString(L, __func__, 1, "channel");
+    const QString message = getVerifiedString(L, __func__, 2, "message");
+
+    Host* pHost = &getHostFromLua(L);
+    if (!pHost->mmcpServer) {
+        pHost->initMMCPServer();
+    }
+
+    QPair<bool, QString> const result = pHost->mmcpServer->chatSideChannel(channel, message);
+
+    if (!result.first) {
+        return warnArgumentValue(L, __func__, result.second.toUtf8().constData());
+    }
+
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+int TLuaInterpreter::chatSnoop(lua_State* L)
+{
+    const QString target = getVerifiedString(L, __func__, 1, "target");
+
+    Host* pHost = &getHostFromLua(L);
+    if (!pHost->mmcpServer) {
+        pHost->initMMCPServer();
+    }
+
+    QPair<bool, QString> const result = pHost->mmcpServer->snoop(target);
+
+    if (!result.first) {
+        return warnArgumentValue(L, __func__, result.second.toUtf8().constData());
+    }
+
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+int TLuaInterpreter::chatStartServer(lua_State* L)
+{
+    int port;
+    const int n = lua_gettop(L);
+    if (n > 1) {
+        port = getVerifiedInt(L, __func__, 2, "port number {default = 4050}", true);
+        if (port > 65535 || port < 1) {
+            return warnArgumentValue(L, __func__, qsl("invalid port number %1 given, if supplied it must be in range 1 to 65535").arg(port));
+        }
+    } else {
+        port = 4050;
+    }
```

Comment:
What is the first argument? It is not even examined here so why bother with it - instead perhaps this code should be:
```suggestion
    int port = 4050;
    if (lua_gettop(L) > 0) {
        port = getVerifiedInt(L, __func__, 1, "port number {default = 4050}", true);
        if (port > 65535 || port < 1) {
            return warnArgumentValue(L, __func__, qsl("invalid port number %1 given, if supplied it must be in range 1 to 65535").arg(port));
        }
    }
```

Reply:
Thx